apps/wakunode2/app.nim

```diff
@@ -587,25 +620,24 @@ proc startNode(node: WakuNode, conf: WakuNodeConf,
 
 proc startApp*(app: var App): AppResult[void] =
 
-  try:
-    (waitFor startNode(app.node,app.conf,app.dynamicBootstrapNodes)).isOkOr:
-      return err(error)
-  except CatchableError:
-    return err("exception starting node: " & getCurrentExceptionMsg())
+  let nodeRes = catch: (waitFor startNode(app.node,app.conf,app.dynamicBootstrapNodes))
+  if nodeRes.isErr():
+    return err("exception starting node: " & nodeRes.error.msg)
+
+  nodeRes.get().isOkOr:
+    return err("exception starting node: " & error)
 
   # Update app data that is set dynamically on node start
   app.updateApp().isOkOr:
     return err("Error in updateApp: " & $error)
 
-  if app.wakuDiscv5.isSome():
-    let wakuDiscv5 = app.wakuDiscv5.get()
-
-    let res = wakuDiscv5.start()
-    if res.isErr():
-      return err("failed to start waku discovery v5: " & $res.error)
+  if (let wakuDiscv5 =  app.wakuDiscv5.get(); app.wakuDiscv5.isSome()):
+    let wakudiscv5Res = catch: (waitFor wakuDiscv5.start())
+    if wakudiscv5Res.isErr():
```

Comment:
Isn't this weird?

First we check `wakudiscv5Res.isErr()`

And the `wakudiscv5Res.get().isOkOr:`

Can't we have just one? Either it errors or not. But checking if getting the error itself errored, seems weird?

Reply:
Good catch! It's weird indeed